include/oneapi/dpl/experimental/kt/esimd_radix_sort.h

```diff
@@ -0,0 +1,84 @@
+// -*- C++ -*-
+//===-- esimd_radix_sort.h --------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------===//
+
+#ifndef _ONEDPL_KT_ESIMD_RADIX_SORT_H
+#define _ONEDPL_KT_ESIMD_RADIX_SORT_H
+
+#include "kernel_param.h"
+#include "../../pstl/utils_ranges.h"
+#include "../../pstl/hetero/dpcpp/utils_ranges_sycl.h"
+#include "../../pstl/hetero/dpcpp/parallel_backend_sycl_utils.h"
+
+#include <cstdint>
+
+#include "internal/esimd_radix_sort_utils.h"
+#include "internal/esimd_radix_sort_dispatchers.h"
+
+namespace oneapi::dpl::experimental::kt::esimd
+{
+
+// TODO: make sure to provide sufficient diagnostic if input does not allow either reading or writing
+template <bool __is_ascending = true, ::std::uint8_t __radix_bits = 8, typename _KernelParam, typename _KeysRng>
+sycl::event
+radix_sort(sycl::queue __q, _KeysRng&& __keys_rng, _KernelParam __param = {})
+{
+    if (__keys_rng.size() < 2)
```

Comment:
oneDPL has such trivial sequence checks on the algorithm pattern (parallel/hetero)  level, for example https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h#L1216

Reply:
I tried moving them at `__impl::__radix_sort` level, but, alas, `oneapi::dpl::__ranges::__get_sycl_range::operator()` does not allow zero-sized ranges due to checks like `assert(__n > 0);` . So this is not a trivial thing.

Reply:
I have noted it in #1311 for further code quality enhancements.